### PR TITLE
Fix bytes parser

### PR DESCRIPTION
**Pull request by Musteblume**
Fix bytes parser to support correct parsing into not zero initialized arrays. Add additional unit tests.

### CODE REVIEW

File: libvast/test/parseable.cpp [hunk 1]
```diff
@@ -474,6 +474,39 @@ TEST(byte) {
...
+  CHECK(a8[3] == 0x04);
+  CHECK(a8[4] == 0x05);
+  CHECK(a8[5] == 0x06);
+  CHECK(a8[6] == 0x07);
+  CHECK(a8[7] == 0x08);
+  auto ip = "\xdf\x00\x0d\xb8\x00\x00\x00\x00\x02\x02\xb3\xff\xfe\x1e\x83\x28"s;
+  f = ip.begin();
+  l = f + 16;
+  std::array<uint8_t, 16> a16;
+  CHECK(bytes<16>(f, l, a16));
```
Comment (mavam) [inline libvast/test/parseable.cpp]: You can ditch `f` and `l` and rewrite this as `CHECK(bytes<16>(ip, a16));`.

Or keep the iterators and add `CHECK(f == l)`. (Then, however, rewrite line 491 as `l = ip.end()`)

File: libvast/test/parseable.cpp [hunk 2]
```diff
@@ -474,6 +474,39 @@ TEST(byte) {
   CHECK(a5[2] == 0x03);
   CHECK(a5[3] == 0x04);
   CHECK(a5[4] == 0x05);
+  f = str.begin();
+  l = f + 8;
+  std::array<uint8_t, 8> a8;
+  CHECK(bytes<8>(f, l, a8));
```
Comment (mavam) [inline libvast/test/parseable.cpp]: (See note below, which also applies here.)

**Comment by Musteblume** [comment]
Done

**Comment by mavam** [comment]
Muchas gracias!

**Pull request merged**